app/controllers/patients_controller.rb

```diff
@@ -466,18 +463,25 @@ def update_status
       end
     end
 
-    # Update patient and all group members
-    if params.permit(:apply_to_household)[:apply_to_household]
-      ([patient] + (current_user.get_patient(patient.responder_id)&.dependents || [])).uniq.each do |member|
-        update_monitoring_fields(member, params, patient[:id] == member[:id] ? :patient : :dependent, :group)
-      end
-      # Update patient and all group members in continuous exposure
-    elsif params.permit(:apply_to_household_cm_only)[:apply_to_household_cm_only] # update patient and group members only with continuous exposure on
-      ([patient] + (current_user.get_patient(patient.responder_id)&.dependents&.where(continuous_exposure: true) || [])).uniq.each do |member|
-        update_monitoring_fields(member, params, patient[:id] == member[:id] ? :patient : :dependent, :group_cm)
-      end
-    else # Update patient
-      update_monitoring_fields(patient, params, :patient, :none)
+    # Update patient
+    update_monitoring_fields(patient, params, :patient, :none)
+
+    # If not applying to household, return
+    apply_to_household_ids = params.permit(apply_to_household_ids: [])[:apply_to_household_ids]
+    return unless params.permit(:apply_to_household)[:apply_to_household] && !apply_to_household_ids.nil?
+
+    # If a household member has been removed, they should not be updated
+    current_household_ids = patient.household.where(purged: false).where.not(id: patient.id).pluck(:id)
+    diff_household_array = apply_to_household_ids - current_household_ids
+    unless diff_household_array.empty?
+      error_message = 'Apply to household action failed: changes have been made to this household. Please refresh.'
+      render(json: { error: error_message }, status: :bad_request) && return
```

Comment:
Could you add a test that tests this error in particular? You can see an example of this kind of test here: https://github.com/SaraAlert/SaraAlert/blob/master/test/controllers/patients_controller_test.rb#L158

Reply:
I did, see here: https://github.com/SaraAlert/SaraAlert/blob/840-apply-to-household-members/test/controllers/patients_controller_test.rb#L949

Reply:
Missed that somehow when reviewing those tests - thanks!